### PR TITLE
fix 'statement has no effect' compiler warnings in logging macros

### PR DESCRIPTION
**Pull request by tommadams**

### CODE REVIEW

File: cc/logging.h [hunk 1]
```diff
@@ -70,6 +72,26 @@ class CheckFailStream {
   LogStream impl_;
 };
 
+// CheckNoOpStream is used in non-debug builds (i.e. when NDEBUG defined) that
+// allows the compiler to completely optimize away MG_DCHECK macros.
```
Comment (amj) [inline cc/logging.h]: any compiler settings required for this optimization?

Comment (tommadams) [reply cc/logging.h]: Clang 7.0 optimizes `MG_DCHECK` away with `-O1` and `-DNDEBUG`.
gcc 7.1 optimizes it away even with `-O0` and `-DNDEBUG`.

https://godbolt.org/z/tq3mr5

**Comment by tommadams** [comment]
`pull-tf-minigo-cc` passes \o/

**Pull request merged**